enhancements/rhcos/rhcos-cloud-marketplace-images.md

```diff
@@ -0,0 +1,330 @@
+  - "@sdodson"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "None"
+creation-date: 2024-07-12
+last-updated: 2024-07-12
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - "SPSTRAT-271"
+see-also:
+replaces:
+superseded-by:
+---
+
+# RHCOS Cloud-Marketplace Images
+
+## Summary
+
+This enhancement proposes a process to upload/publish
+cloud (AWS, Azure, & GCP) marketplace images and include
+the image details in the RHCOS stream.
+
+## Motivation
+
+Publishing marketplace versions in the RHCOS stream will
+improve user experience and relieve tech debt:
+
+* Marketplace images published as part of Project Stratosphere are incomplete, requiring users to use
+boot images from previous releases, which slows
+down machine provisioning.
+* As marketplace images are not included in the coreos
+stream, users must manually enter the image details.
+With these changes, the experience will be improved
+so that users can simply indicate they want to use
+marketplace images and the installer will select the
+appropriate image.
+* Publishing a free Azure marketplace image would
+significantly decrease Azure installation times &
+would unblock Azure hosted control planes. On Azure, marketplace
+images are the only globally available, production-ready API for
+publishing images. Currently, every install creates a managed image
+utilizing the 16GB VHD file found in the RHCOS stream. Publishing
+an image would immediately save time during installs, reduce
+installer code maintenance, and reduce cluster cost.
+
+### User Stories
+
+* As an OpenShift cluster admin, I want to specify
+a boolean value for marketplace images in the install
+config so that I can utilize marketplace images without
+looking up specific version numbers.
+
+### Goals
+
+* A well-defined process for publishing cloud marketplace
+images and including them in the RHCOS stream.
+* Setup non-paid Azure marketplace offering: determine publisher
+* Definition in RHCOS stream to separate multiple similar offers (e.g. ROSA-specific marketplace images vs.
+OpenShift AWS marketplace images, Azure paid vs. free images)
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+This enhancement proposes a process for publishing the images and
+a specification for how the images would be published in the stream.
+
+The process for uploading images will be based on RHCOS stream bump
+PRs. New PRs to the installer will trigger a request to upload images.
+When the images are available, the PR will be merged to update the stream.
+
+Marketplace images will be included in the
+`.architectures[("aarch64","x86_64")].rhel_coreos_extensions` fields. 
+
+
+### Workflow Description
```

Comment:
This describe the "older" workflow, not the one I suggested in https://miro.com/app/board/uXjVK0sf6No=/, which is more linear and does not need us to do multiple things in a single PR.

Reply:
Thanks for bringing this back up. I didn't understand the Alternate Workflow, but after reviewing it a bit more, I understand a bit better: 

Marketplace images are contained in a separate rhcos-marketplace.json file. Do you have a sense of how that file is generated? By RHCOS team or the installer team?

Reply:
We generate the rhcos.json file with our tooling. Ideally the new `rhcos-marketplace.json` file is generated by the team responsible for uploading the images, using the same format.

Reply:
I think it's https://github.com/coreos/stream-metadata-go

Reply:
@travier I investigated the idea of using multiple files, but landed on the current proposal as a simpler way to solve the same problem. I addressed multiple files in the `Alternatives` section. Would you please take a fresh look?

The idea is really quite simple, but I hope I have not made it too confusing when trying to explain.